x/stake/keeper_test.go

```diff
@@ -117,8 +117,8 @@ func TestValidatorBasics(t *testing.T) {
 	resVals = keeper.GetValidatorsBonded(ctx)
 	require.Equal(t, 3, len(resVals))
 	assert.True(ValEq(t, validators[0], resVals[2])) // order doesn't matter here
-	assert.True(ValEq(t, validators[1], resVals[0]))
-	assert.True(ValEq(t, validators[2], resVals[1]))
+	assert.True(ValEq(t, validators[1], resVals[1]))
+	assert.True(ValEq(t, validators[2], resVals[0]))
```

Comment:
I am Not sure why the order doesn’t matter here, but weirdly I had to change the order after updating the way validators are created to the function that makes 100 of them, rather than the hardcoded 10 there were before. After looking for a while I could not figure out why I had to change their order, however the test still works. So hopefully this is exactly how it should be and I just couldn't figure out why resVals got its order switched around

Reply:
The bonded validator store sorts by Address, the only thing that matters is that records are in this store... The only store which the key record matters is the ValidatorsByPower store. The order could have gotten switched around if the addresses used in the test where changed